### PR TITLE
Formatting changes

### PR DESCRIPTION
**Pull request by EmilyBourne**

### CODE REVIEW

File: fullSimulation.py [hunk 1]
```diff
@@ -1,21 +1,25 @@
-from mpi4py                 import MPI
+from pygyro.diagnostics.diagnostic_collector import DiagnosticCollector
+from pygyro.utilities.savingTools import setupSave
+from pygyro.poisson.poisson_solver import DensityFinder, QuasiNeutralitySolver
+from pygyro.advection.advection import FluxSurfaceAdvection, VParallelAdvection, PoloidalAdvection, ParallelGradient
+from pygyro.initialisation.setups import setupCylindricalGrid, setupFromFile
+from pygyro.model.grid import Grid
+from pygyro.model.layout import LayoutSwapper, getLayoutHandler
```
Comment (EmilyBourne) [inline fullSimulation.py:7]: @dominikbell Normally all python built-in imports should come first. Do you know why this was re-ordered?

File: moduleGenerator.py [hunk 2]
```diff
@@ -1,36 +1,51 @@
...
+                               'double', 'double[:]', 'int', 'double[:]', 'int'])
+spline_context.insert_function(eval_spline_1d_vector, [
+                               'double', 'double[:]', 'int', 'double[:]', 'double[:]', 'int'])
+
+init_context = ContextPyccel(
+    name='initialiser_funcs', context_folder='pygyro.initialisation', output_folder='pygyro.advection')
+init_context.insert_function(
+    f_eq, ['double', 'double', 'double', 'double', 'double', 'double', 'double', 'double', 'double'])
+
+aas_func = epyccel(pygyro.advection.accelerated_advection_steps,
```
Comment (EmilyBourne) [inline moduleGenerator.py:50]: This file can be deleted. It was needed for pyccel in early 2018 but was already obsolete by the summer

File: pygyro/advection/accelerated_advection_steps.py [hunk 3]
```diff
@@ -52,141 +52,150 @@ def poloidal_advection_step_expl( f, dt, v, rPts, qPts,
...
             # Step two of Heun method
             # x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
-            endPts_k2_q[i,j] = (qPts[i] - (drPhi_0[i,j]     + drPhi_k[i,j])*multFactor_half) % (2*pi)
-            if endPts_k2_q[i,j]<0: # Needed for C due to pyccel issue #854
-                endPts_k2_q[i,j] += 2*pi
-            endPts_k2_r[i,j] = rPts[j] + (dthetaPhi_0[i,j] + dthetaPhi_k[i,j])*multFactor_half
+            endPts_k2_q[i, j] = (
+                qPts[i] - (drPhi_0[i, j] + drPhi_k[i, j])*multFactor_half) % (2*pi)
+            if endPts_k2_q[i, j] < 0:  # Needed for C due to pyccel issue #854
+                endPts_k2_q[i, j] += 2*pi
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:89]: Obsolete comment. This should now be handled by the `%` operator

File: fullSimulation.py (hunk 1 above)
Comment (dominikbell) [reply fullSimulation.py:7]: I am not sure, I reordered them now and it didn't change it upon formatting ..

File: moduleGenerator.py (hunk 2 above)
Comment (dominikbell) [reply moduleGenerator.py:50]: ok, will do

File: pygyro/advection/accelerated_advection_steps.py [hunk 4]
```diff
@@ -52,141 +52,150 @@ def poloidal_advection_step_expl( f, dt, v, rPts, qPts,
...
-@allow_negative_index('vals') # Needed for C due to pyccel issue #854
-def get_lagrange_vals(i,shifts,vals,qVals,thetaShifts,kts,deg,coeffs):
+        for i, v in enumerate(vPts):
+            while (v < vMin):
+                v += vDiff
+            while (v > vMax):
+                v -= vDiff
+            f[i] = eval_spline_1d_scalar(v, kts, deg, coeffs, 0)
+
+# @allow_negative_index('vals') # Needed for C due to pyccel issue #854
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:159]: I wonder if it would be better to leave this activated and change line 176 to `vals[i-s, k, j] ...`?

File: pygyro/advection/accelerated_advection_steps.py [hunk 5]
```diff
@@ -223,95 +234,96 @@ def poloidal_advection_step_impl( f, dt, v, rPts, qPts,
...
             for j in range(nPts_r):
                 # Handle theta boundary conditions
-                while (endPts_k1_q[i,j]<0):
-                    endPts_k1_q[i,j]+=2*pi
-                while (endPts_k1_q[i,j]>2*pi):
-                    endPts_k1_q[i,j]-=2*pi
+                while (endPts_k1_q[i, j] < 0):
+                    endPts_k1_q[i, j] += 2*pi
+                while (endPts_k1_q[i, j] > 2*pi):
+                    endPts_k1_q[i, j] -= 2*pi
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:254]: These can be replaced with modulo operators

File: pygyro/advection/accelerated_advection_steps.py [hunk 6]
```diff
@@ -223,95 +234,96 @@ def poloidal_advection_step_impl( f, dt, v, rPts, qPts,
...
-                    norm=diff
-                diff=abs(endPts_k2_r[i,j]-endPts_k1_r[i,j])
-                if (diff>norm):
-                    norm=diff
-                endPts_k1_q[i,j]=endPts_k2_q[i,j]
-                endPts_k1_r[i,j]=endPts_k2_r[i,j]
+                endPts_k2_q[i, j] = (
+                    qPts[i] - (drPhi_0[i, j] + drPhi_k[i, j])*multFactor) % (2*pi)
+                if endPts_k2_q[i, j] < 0:  # Needed for C due to pyccel issue #854
+                    endPts_k2_q[i, j] += 2*pi
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:280]: Modulo operator now works

File: pygyro/advection/accelerated_advection_steps.py [hunk 7]
```diff
@@ -223,95 +234,96 @@ def poloidal_advection_step_impl( f, dt, v, rPts, qPts,
...
-                        endPts_k2_q[i,j]-=2*pi
-                    while (endPts_k2_q[i,j]<0):
-                        endPts_k2_q[i,j]+=2*pi
-                    f[i,j]=eval_spline_2d_scalar(endPts_k2_q[i,j],endPts_k2_r[i,j],
-                                                        kts1Pol, deg1Pol, kts2Pol, deg2Pol,
-                                                        coeffsPol,0,0)
+                    while (endPts_k2_q[i, j] > 2*pi):
+                        endPts_k2_q[i, j] -= 2*pi
+                    while (endPts_k2_q[i, j] < 0):
+                        endPts_k2_q[i, j] += 2*pi
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:310]: Modulo operator

File: pygyro/advection/accelerated_advection_steps.py [hunk 8]
```diff
@@ -223,95 +234,96 @@ def poloidal_advection_step_impl( f, dt, v, rPts, qPts,
...
-                        endPts_k2_q[i,j]-=2*pi
-                    while (endPts_k2_q[i,j]<0):
-                        endPts_k2_q[i,j]+=2*pi
-                    f[i,j]=eval_spline_2d_scalar(endPts_k2_q[i,j],endPts_k2_r[i,j],
-                                                kts1Pol, deg1Pol, kts2Pol, deg2Pol, coeffsPol,0,0)
-
+                    while (endPts_k2_q[i, j] > 2*pi):
+                        endPts_k2_q[i, j] -= 2*pi
+                    while (endPts_k2_q[i, j] < 0):
+                        endPts_k2_q[i, j] += 2*pi
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:327]: Modulo operator

File: pygyro/advection/accelerated_advection_steps.py (hunk 7 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py:310]: will do

File: pygyro/advection/advection.py [hunk 9]
```diff
@@ -448,85 +474,87 @@ def step( self, f: np.ndarray, dt: float, phi: Spline2D, v: float ):
...
             else:
-                while (theta>2*pi):
-                    theta-=2*pi
-                while (theta<0):
-                    theta+=2*pi
-                return self._spline.eval(theta,r)
+                while (theta > 2*pi):
+                    theta -= 2*pi
+                while (theta < 0):
+                    theta += 2*pi
```
Comment (EmilyBourne) [inline pygyro/advection/advection.py:556]: Modulo

Comment (dominikbell) [reply pygyro/advection/advection.py:556]: yes

File: pygyro/advection/accelerated_advection_steps.py (hunk 5 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py:254]: yes

**Comment by EmilyBourne** [comment]
If you can find the name of the format you used we might be able to add a linter to check that the format is respected

File: pygyro/advection/accelerated_advection_steps.py (hunk 6 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py:280]: The problem was that the modulo operator was not implemented correctly; now these lines are in general unnecessary.

File: pygyro/advection/accelerated_advection_steps.py (hunk 3 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py:89]: same as below

**Comment by EmilyBourne** [comment]
We seem to have lost a symbolic link to splines in the advection folder (necessary for numba)

File: pygyro/initialisation/initialiser_funcs.py [hunk 10]
```diff
@@ -1,75 +1,106 @@
-from pyccel.decorators import types, pure
-
-@pure
```
Comment (EmilyBourne) [inline pygyro/initialisation/initialiser_funcs.py:3]: Why did you remove the `@pure` decorator? This is useful for the fortran compiler to make optimisations. See https://github.com/pyccel/pyccel/blob/master/tutorial/decorators.md#pure

File: pygyro/initialisation/pythran_initialiser_funcs.py [hunk 11]
```diff
@@ -1,55 +1,112 @@
...
-        for j,v in enumerate(vPar):
-            surface[i,j]=f_eq(r,v,CN0,kN0,deltaRN0,rp,Cti,kti,deltaRti)*(1+eps*perturbation(r,q,z,m,n,rp,deltaR,R0))
-
-#pythran export feq_vector(float64[:,:],float64[:],float64[:],float64,float64,float64,float64,float64,float64,float64)
-def feq_vector(surface,r_vec,vPar,CN0,kN0,deltaRN0,rp,Cti,kti,deltaRti):
-    for i,r in enumerate(r_vec):
-        for j,v in enumerate(vPar):
-            surface[i,j]=f_eq(r,v,CN0,kN0,deltaRN0,rp,Cti,kti,deltaRti)
+# pythran export n0(float64,float64,float64,float64,float64)
+# pythran export n0(float64[:],float64,float64,float64,float64)
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pygyro/initialisation/pythran_initialiser_funcs.py:4]: I don't know if the space is accepted. This needs checking

File: pygyro/advection/accelerated_advection_steps.py [hunk 12]
```diff
@@ -1,19 +1,18 @@
-from pyccel.decorators  import types, pure, allow_negative_index
+from pyccel.decorators import allow_negative_index
 from ..splines.spline_eval_funcs import eval_spline_2d_cross, eval_spline_2d_scalar, eval_spline_1d_scalar
-from ..initialisation.initialiser_funcs               import f_eq
-
-@pure
```
Comment (EmilyBourne) [inline pygyro/advection/accelerated_advection_steps.py:5]: Again why is `@pure` removed?

File: pygyro/advection/advection.py [hunk 13]
```diff
@@ -448,85 +474,84 @@ def step( self, f: np.ndarray, dt: float, phi: Spline2D, v: float ):
...
-                            phiBases[1].knots, phi.coeffs,
-                            phiBases[0].degree, phiBases[1].degree,
-                            polBases[0].knots, polBases[1].knots,
-                            self._spline.coeffs, polBases[0].degree,
-                            polBases[1].degree, self._constants.CN0,
-                            self._constants.kN0, self._constants.deltaRN0,
-                            self._constants.rp, self._constants.CTi,
-                            self._constants.kTi, self._constants.deltaRTi,
-                            self._constants.B0, self._nulEdge)
+            poloidal_advection_step_expl(f, float(dt), v, self._points[1],
```
Comment (EmilyBourne) [inline pygyro/advection/advection.py:484]: Surely the implicit advection also has this problem? I'm surprised it doesn't appear in other operators. Wouldn't it be safer to cast in `fullSimulation.py`?

File: pygyro/tools/getSlice.py [hunk 14]
```diff
@@ -1,33 +1,34 @@
-from mpi4py                 import MPI
+from pygyro.initialisation.setups import setupFromFile
+import h5py
+from mpi4py import MPI
 import time
 setup_time_start = time.clock()
 
-import h5py
-#~ import cProfile, pstats, io
+# ~ import cProfile, pstats, io
```
Comment (EmilyBourne) [inline pygyro/tools/getSlice.py:7]: Maybe we can remove the tildes? They were inserted by my old editor which commented with `#~` for some reason

File: pygyro/tools/getPhiSlice.py [hunk 15]
```diff
@@ -1,26 +1,29 @@
-from mpi4py                 import MPI
+from .. import splines as spl
+from ..initialisation import constants
+from ..model.process_grid import compute_2d_process_grid
+from ..model.grid import Grid
+from ..model.layout import LayoutSwapper
```
Comment (EmilyBourne) [inline pygyro/tools/getPhiSlice.py:5]: Again, strange import order. Doesn't codacy/pylint complain about this?

File: pygyro/splines/spline_eval_funcs.py [hunk 16]
```diff
@@ -1,9 +1,4 @@
-from pyccel.decorators  import types, pure, stack_array
```
Comment (EmilyBourne) [inline pygyro/splines/spline_eval_funcs.py:1]: Why is `@pure` and worse `@stack_array` removed?!

File: pygyro/advection/accelerated_advection_steps.py (hunk 6 above)
Comment (EmilyBourne) [reply pygyro/advection/accelerated_advection_steps.py]: It seems there is a bug with `%=` you will need `a = a%(2*pi)`

Comment (EmilyBourne) [reply pygyro/advection/accelerated_advection_steps.py]: I have opened an issue in pyccel https://github.com/pyccel/pyccel/issues/1172

File: pygyro/splines/spline_eval_funcs.py (hunk 16 above)
Comment (dominikbell) [reply pygyro/splines/spline_eval_funcs.py:1]: They will be readded.

File: pygyro/advection/accelerated_advection_steps.py (hunk 4 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py:151]: Apparently, this leads to a `Aborted (core dumped)` error. I will open an issue on the `pyccel` GitHub about this.

File: pygyro/initialisation/initialiser_funcs.py (hunk 10 above)
Comment (dominikbell) [reply pygyro/initialisation/initialiser_funcs.py:3]: Yes, I will readd them

File: pygyro/advection/accelerated_advection_steps.py (hunk 12 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py:5]: Again, I will readd them

File: pygyro/tools/getSlice.py (hunk 14 above)
Comment (dominikbell) [reply pygyro/tools/getSlice.py:7]: done

File: pygyro/tools/getPhiSlice.py (hunk 15 above)
Comment (dominikbell) [reply pygyro/tools/getPhiSlice.py:5]: No, but I changed the order.

File: pygyro/initialisation/pythran_initialiser_funcs.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (dominikbell) [reply pygyro/initialisation/pythran_initialiser_funcs.py:4]: I don't know. I readded them, but every formatting of this file will revert this.

**Comment by EmilyBourne** [comment]
Codacy errors also need fixing before I can merge

File: pygyro/advection/accelerated_advection_steps.py (hunk 6 above)
Comment (dominikbell) [reply pygyro/advection/accelerated_advection_steps.py]: Great, I replaced it by `%(2*pi)`

**Comment by dominikbell** [comment]
> If you can find the name of the format you used we might be able to add a linter to check that the format is respected

the standard is called PEP8 which is the [default format in VSC](https://docs.microsoft.com/en-us/visualstudio/python/formatting-python-code?view=vs-2022).

File: pygyro/advection/accelerated_advection_steps.py (hunk 4 above)
Comment (EmilyBourne) [reply pygyro/advection/accelerated_advection_steps.py]: This is now fixed in the master branch of pyccel. It will be in the next version

File: pygyro/initialisation/pythran_initialiser_funcs.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pygyro/initialisation/pythran_initialiser_funcs.py]: I checked and pythran doesn't seem to care if there is a space so we can leave the formatting with a space

File: pygyro/utilities/grid_plotter.py [hunk 17]
```diff
@@ -562,7 +562,7 @@ def checkProgress(self):
                 self.buttons[1].set_active(True)
             else:
                 self.prepare_data_reception()
-            for i in range(len(self.completedRanks)):
+            for i, x in enumerate(self.completedRanks):
```
Comment (EmilyBourne) [inline pygyro/utilities/grid_plotter.py:565]: Unused variable needs: `for i,_ in enumerate(self.completedRanks):`
